src/Features/LanguageServer/ProtocolUnitTests/UriTests.cs

```diff
@@ -79,28 +79,43 @@ public async Task TestWorkspaceDocument_WithFileScheme(bool mutatingLspWorkspace
     {
         var documentFilePath = @"C:\A.cs";
         var markup =
-@$"<Workspace>
-    <Project Language=""C#"" Name=""CSProj1"" CommonReferences=""true"" FilePath=""C:\CSProj1.csproj"">
-        <Document FilePath=""{documentFilePath}"">
-            public class A
-            {{
-            }}
-        </Document>
-    </Project>
-</Workspace>";
+            $$"""
+            <Workspace>
+                <Project Language="C#" Name="CSProj1" CommonReferences="true" FilePath="C:\CSProj1.csproj">
+                    <Document FilePath="{{documentFilePath}}">
+                        public class A
+                        {
+                        }
+                    </Document>
+                </Project>
+            </Workspace>
+            """;
         await using var testLspServer = await CreateXmlTestLspServerAsync(markup, mutatingLspWorkspace);
 
         var workspaceDocument = testLspServer.TestWorkspace.CurrentSolution.Projects.Single().Documents.Single();
         var expectedDocumentUri = ProtocolConversions.CreateAbsoluteUri(documentFilePath);
 
         await testLspServer.OpenDocumentAsync(expectedDocumentUri).ConfigureAwait(false);
 
-        // Verify file is added to the misc file workspace.
-        var (workspace, _, document) = await testLspServer.GetManager().GetLspDocumentInfoAsync(new LSP.TextDocumentIdentifier { Uri = expectedDocumentUri }, CancellationToken.None);
-        Assert.False(workspace is LspMiscellaneousFilesWorkspace);
-        AssertEx.NotNull(document);
-        Assert.Equal(expectedDocumentUri, document.GetURI());
-        Assert.Equal(documentFilePath, document.FilePath);
+        // Verify file is not added to the misc file workspace.
+        {
+            var (workspace, _, document) = await testLspServer.GetManager().GetLspDocumentInfoAsync(new LSP.TextDocumentIdentifier { Uri = expectedDocumentUri }, CancellationToken.None);
+            Assert.False(workspace is LspMiscellaneousFilesWorkspace);
+            AssertEx.NotNull(document);
+            Assert.Equal(expectedDocumentUri, document.GetURI());
+            Assert.Equal(documentFilePath, document.FilePath);
+        }
+
+        // Try again, this time with a uri with different case sensitivity.  This is supported, and is needed by Xaml.
```

Comment:
to clarify - its needed by all files - C# devkit is adding them as upper case drive letters (but the URIs from the client are lowercase drive letters).